server/controllers/authController.ts

```diff
@@ -1,11 +1,57 @@
-import catchAsync from '../utils/catchAsync';
+import { Request, Response, NextFunction } from "express";
+import catchAsync from "../utils/catchAsync";
+import * as msal from "@azure/msal-node";
 
-const login = catchAsync((req, res, next) => {
-  req;
-  res;
-  next;
-});
-
-export default {
-  login,
+const config = {
+  auth: {
+    clientId: `${process.env.CLIENT_ID}`,
+    authority: `${process.env.AUTHORITY_URL}`,
+    clientSecret: `${process.env.CLIENT_SECRET}`,
+  },
+  system: {
+    loggerOptions: {
+      loggerCallback(_loglevel: any, message: string, _containsPii: any) {
+        console.log(message);
+      },
+      piiLoggingEnabled: false,
+      logLevel: msal.LogLevel.Verbose,
+    },
+  },
 };
+
+const loggingSession = new msal.ConfidentialClientApplication(config);
+
+export const login = catchAsync(
+  async (req: Request, res: Response, next: NextFunction) => {
+    const authCodeUrlParameters = {
+      scopes: ["user.read"],
+      redirectUri: "http://localhost:8080/",
```

Comment:
Is it required? It should be dynamic e.g. after login we want to open details of tournament (<url>?next=<path to open after login>)
Maybe we can do it on the frontend side? FE supports `next` param in login page

Reply:
redirectUri is static, set on the Azure AD website. Can't be changed dynamically. Of course this URi can be changed on Azure AD website. It;s just for nowe